client/src/components/Address/EditAddress.js

```diff
@@ -11,12 +13,16 @@ const EditAddress = () => {
   let [city, setCity] = useState(null);
   let [pin_code, setPin_code] = useState(null);
   let [state, setState] = useState(null);
-  let [country_id, setCountry_id] = useState(null);
+  let [country, setCountry] = useState(null);
   let [user_id, setUser_id] = useState(1);
-  let [countries, setCountries] = useState(null);
   const [message, setMessage] = useState(null);
   const navigate = useNavigate();
   let id = window.location.pathname.substring(14, 15);
+  const options = useMemo(() => countryList().getData(), []);
```

Comment:
why are yo using useMemo here?

Reply:
The useMemo  is a local call,  immediately renders value.which has only something to do with this component. when  some value is more urgent that time we can use this.
here we need country list soon so we r using usememo ,we cant use useEffect or useState.

The useEffect is a collective call, async or not, it's collected after all components are rendered